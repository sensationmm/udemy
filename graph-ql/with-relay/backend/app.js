const express = require('express');
const expressGraphQL = require('express-graphql');
const { GraphQLSchema } = require('graphql');
const mongoose = require('mongoose');

const Post = require('./src/types/Post');
const Query = require('./src/types/Query');
const Mutation = require('./src/types/Mutation');

const schema = new GraphQLSchema({
  query: Query,
  mutation: Mutation
})

const app = express();

const MONGO_URI = 'mongodb://kevin:asprilla1@ds119150.mlab.com:19150/with-relay'
mongoose.Promise = global.Promise;
mongoose.connect(MONGO_URI, { useNewUrlParser: true });
mongoose.connection
  .once('open', () => console.log('Connected to MongoLab instance.'))
  .on('error', error => console.log('Error connecting to MongoLab:', error));

app.use('/backend', expressGraphQL({
  schema,
  graphiql: true
}));

app.listen(5000);
console.log('Server started on port 5000');
