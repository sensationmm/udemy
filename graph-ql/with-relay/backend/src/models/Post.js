const mongoose = require('mongoose');
const { Schema } = mongoose;

const PostSchema = new Schema({
  title: String,
  content: String
});

const PostModel = mongoose.model('Post', PostSchema);

module.exports = {
  getPosts: () => {
    return PostModel.find();
  },
  getPost: id => {
    return PostModel.findOne({ _id: id });
  },
  createPost: post => {
    return PostModel(post).save();
  }
};
