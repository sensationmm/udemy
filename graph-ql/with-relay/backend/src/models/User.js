const mongoose = require('mongoose');
const { Schema } = mongoose;

const UserSchema = new Schema({
  username: String,
  password: String,
  fullname: String
});

const UserModel = mongoose.model('User', UserSchema);

module.exports = {
  getUser: (id, username) => {
    return UserModel.findOne({ 
      $or: [{ _id: id }, { username: username }]
    });
  },
  loginUser: loginUserInput => {
    const { username, password } = loginUserInput;
    return UserModel.findOne({
      $and: [{ username: username }, { password: password } ]
    });
  },
  createUser: user => {
    return UserModel(user).save();
  }
};
