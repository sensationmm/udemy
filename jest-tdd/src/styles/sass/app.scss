@import './modules/reset';
@import './modules/colors';

body {
  font-family: arial;
  background: $background;
  padding: 30px;
  font-size: 14px;
  text-align: center;
}

.body {
  width: 500px;
  padding: 20px;
  background: $backgroundContent;
  border: 1px solid $border;
  margin: 0 auto;
}

h1 {
  text-align: center;
}

@media screen and (max-width: 360px) {
  body {
    padding: 10px;
  }
}
