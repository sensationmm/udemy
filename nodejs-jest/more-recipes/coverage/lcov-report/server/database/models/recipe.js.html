<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for server/database/models/recipe.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">server/database/models</a> recipe.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { updateRecipeAttributes } from '../../helpers';
&nbsp;
module.exports = (sequelize, DataTypes) =&gt; {
  const Recipe = sequelize.define('Recipe', {
    id: {
      allowNull: false,
      primaryKey: true,
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4
    },
    userId: {
      type: DataTypes.UUID,
      references: {
        model: 'Users',
        key: 'id'
      }
    },
    title: DataTypes.STRING,
    description: DataTypes.STRING,
    imageUrl: DataTypes.STRING,
    timeToCook: DataTypes.INTEGER,
    ingredients: DataTypes.TEXT,
    procedure: DataTypes.TEXT
  });
&nbsp;
  Recipe.associate = (models) =&gt; {
    Recipe.belongsTo(models.User, {
      foreignKey: 'userId',
      onDelete: 'CASCADE'
    });
    Recipe.hasMany(models.Review, {
      foreignKey: 'recipeId'
    });
  };
&nbsp;
  Recipe.addHook('afterFind', async (results) =&gt; {
    <span class="missing-if-branch" title="else path not taken" >E</span>if (Array.isArray(results)) {
      await Promise.all(results
        .map(<span class="fstat-no" title="function not covered" >as</span>ync sequelizeRecipe =&gt; <span class="cstat-no" title="statement not covered" >updateRecipeAttributes(sequelizeRecipe))</span>);
    } else {
<span class="cstat-no" title="statement not covered" >      return updateRecipeAttributes(results);</span>
    }
  });
&nbsp;
  return Recipe;
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 15:56:30 GMT+0000 (Greenwich Mean Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
