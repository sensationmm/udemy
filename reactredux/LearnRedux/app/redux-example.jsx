var redux = require('redux');

console.log('starting redux example');

var reducer = (state = {name: 'Anonymous'}, action) => {

	switch(action.type) {
		case 'CHANGE_NAME':
			return {
				...state,
				name: action.name
			}
			break;
		default:
			return state;
	}

	return state;
};

var store = redux.createStore(reducer);

console.log('currentState', store.getState());

store.dispatch({
	type: 'CHANGE_NAME',
	name: 'Kevin'
});

console.log('newState', store.getState());