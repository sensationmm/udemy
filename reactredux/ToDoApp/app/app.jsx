var React = require('react');
var ReactDOM = require('react-dom');
var {Route, Router, IndexRoute, hashHistory} = require('react-router');

var ToDoApp = require('ToDoApp');

//Load Foundation
$(document).foundation();

//App css
require('style!css!sass-loader!applicationStyles');

ReactDOM.render(
	<ToDoApp />,
	document.getElementById('app')
);


