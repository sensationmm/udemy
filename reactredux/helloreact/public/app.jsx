var React = require('react');
var ReactDOM = require('react-dom');
var Greeter = require('Greeter');


var firstName = 'Kevin';
var messageString = 'This is from a component!';

ReactDOM.render(
	<Greeter name={firstName} />,
	document.getElementById('app')
);


