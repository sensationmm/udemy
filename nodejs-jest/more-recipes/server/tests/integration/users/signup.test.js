import supertest from 'supertest';
import faker from 'faker';

import { User } from '../../../database/models';

import app from '../../../index';

describe('User sign up', () => {
  test('should register a new user', async () => {

    const fakeUser = {
      name: 'Test User',
      email: faker.internet.email(),
      password: 'password'
    };

    const res = await supertest(app).post('/api/v1/users/signup').send(fakeUser);

    expect(res.status).toBe(200);
    expect(res.body.data.user.name).toEqual(fakeUser.name);
    expect(res.body.data.user.email).toEqual(fakeUser.email);
    expect(res.body.data.access_token).toBeTruthy();

    const userFromDatabase = await User.find({ where: { email: fakeUser.email }});
    expect(userFromDatabase).toBeTruthy();
  });

  test('should return validation error for duplicate email', async () => {
    const fakeUser = {
      name: 'Test User',
      email: faker.internet.email(),
      password: 'password'
    };

    await User.create(fakeUser);

    const res = await supertest(app).post('/api/v1/users/signup').send(fakeUser);

    expect(res.status).toBe(422);
    expect(res.body).toMatchSnapshot();
  });

});
