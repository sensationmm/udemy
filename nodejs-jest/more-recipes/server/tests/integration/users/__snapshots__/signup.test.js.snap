// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User sign up should return validation error for duplicate email 1`] = `
Object {
  "data": Object {
    "errors": Array [
      "A user with this email already exists.",
    ],
  },
  "status": "fail",
}
`;
